src/include/catalog/catalog_entry/function_catalog_entry.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "catalog_entry.h"
+#include "function/scalar_function.h"
+
+namespace kuzu {
+namespace catalog {
+
+class FunctionCatalogEntry : public CatalogEntry {
+public:
+    //===--------------------------------------------------------------------===//
+    // constructors
+    //===--------------------------------------------------------------------===//
+    FunctionCatalogEntry() = default;
+
+    FunctionCatalogEntry(
+        CatalogEntryType entryType, std::string name, function::function_set functionSet);
+
+    //===--------------------------------------------------------------------===//
+    // getters & setters
+    //===--------------------------------------------------------------------===//
+    function::function_set& getFunctionSet() { return functionSet; }
+
+    //===--------------------------------------------------------------------===//
+    // serialization & deserialization
+    //===--------------------------------------------------------------------===//
+    void serialize(common::Serializer& /*serializer*/) const override { return; }
```

Comment:
Comment why we never serialize FunctionCatalogEntry. My understanding is that we always register when creating Catalog instead of deserializing from file.